restructure V1 migration guide (#950)
Deltares/hydromt: docs/dev/migrating-to-v1.rst

| MeshModel.mesh        | MeshComponent       | Component for managing unstructured grids              |
+-----------------------+---------------------+--------------------------------------------------------+
| VectorModel.vector    | VectorComponent     | Component for managing geospatial vector data          |
+-----------------------+---------------------+--------------------------------------------------------+

Changes to the `yaml` HydroMT configuration file format
-------------------------------------------------------

**Rationale**

Given the changes to the `Model` class and new `ModelComponent` the `yaml` configuration file format has
also been updated. The new format is more flexible and allows configuration of the model object and its
components. Components can be added during instantiation by defining them in the yaml which is useful to
define a model on-the-fly based on existing components. Note that this is however not necessary for plugins
as the available components are defined in their implementation of the `Model` class.

**Changes required**

The first change to the YAML format is that now, at the root of the documents are three keys:

> - `modeltype` (optional) details what kind of model is going to be used in the model.

Author: @hboisgon @deltamarnix In my opinion `model_name` would be better. A type refers to a category of models, but here you want the actual model software name right?

```suggestion
- `model_name` (optional) the name of the model software.
```

Author: In addition, I think it would be nice to have a `Model.from_yml` class method to move the CLI functionality to the model class. In that way also python users can benefit from the yaml format. Something for v1-beta?
